### DISCUSSION TITLE
mini.align: How to add more spaces between two columns

### DISCUSSION THREAD

**Discussion by rafamadriz** [nvim-mini/mini.nvim#313] [Q&A] [answered]
Example I have this:

```sh
setopt EXTENDED_HISTORY       # Write the history file in the ":start:elapsed;command" format.
setopt INC_APPEND_HISTORY     # Write to the history file immediately, not when the shell exits.
setopt SHARE_HISTORY          # Share history between all sessions.
setopt HIST_EXPIRE_DUPS_FIRST # Expire duplicate entries first when trimming history.
setopt HIST_IGNORE_DUPS       # Don't record an entry that was just recorded again.
setopt HIST_IGNORE_ALL_DUPS   # Delete old recorded entry if new entry is a duplicate.
setopt HIST_FIND_NO_DUPS      # Do not display a line previously found.
setopt HIST_IGNORE_SPACE      # Don't record an entry starting with a space.
setopt HIST_SAVE_NO_DUPS      # Don't write duplicate entries in the history file.
setopt HIST_REDUCE_BLANKS     # Remove superfluous blanks before recording entry.
setopt HIST_VERIFY            # Don't execute immediately upon history expansion.
```

I want this:

```sh
setopt EXTENDED_HISTORY             # Write the history file in the ":start:elapsed;command" format.
setopt INC_APPEND_HISTORY           # Write to the history file immediately, not when the shell exits.
setopt SHARE_HISTORY                # Share history between all sessions.
setopt HIST_EXPIRE_DUPS_FIRST       # Expire duplicate entries first when trimming history.
setopt HIST_IGNORE_DUPS             # Don't record an entry that was just recorded again.
setopt HIST_IGNORE_ALL_DUPS         # Delete old recorded entry if new entry is a duplicate.
setopt HIST_FIND_NO_DUPS            # Do not display a line previously found.
setopt HIST_IGNORE_SPACE            # Don't record an entry starting with a space.
setopt HIST_SAVE_NO_DUPS            # Don't write duplicate entries in the history file.
setopt HIST_REDUCE_BLANKS           # Remove superfluous blanks before recording entry.
setopt HIST_VERIFY                  # Don't execute immediately upon history expansion.
```

I know how to do the reverse of that, instead of adding spaces trimming them, in this case with the key sequence: `gAip=#<CR>` will align the second column with `#` (Taking me from the second example to the first one). But I can't figure how to add spaces between tow columns once they have been aligned.

**Comment by echasnovski** [comment, marked as answer]
The cleanest approach for me seems to not do this within 'mini.align'. You align with `gaip#` leading to the first example (or shifted right part depending on the initial text) and then use Visual-block mode to tweak spacing:
- Place cursor on the `#` in the first line.
- Press `<C-v>` (Ctrl + v).
- Position cursor on the `#` in the last line. This should lead to a single column selection as `#`s are aligned.
- Press `I` and add as many spaces as you'd like.
- Press `<Esc>`. This will exit Visual-block mode and add whitespace to all selected lines.

If you **must** do this with 'mini.align', there are these steps that work in this particular instance:
- `gAip` to start align with preview.
- `s` followed by `#.*` followed by `<CR>` to align by `#.*` pattern.
- `t` to trim whitespace.
- `m` followed by any number of whitespace you'd like followed by `<CR>` to merge with input number of whitespaces.
- `<CR>` to confirm alignment.

**Reply by rafamadriz** [reply]
Thanks for taking the time to answer, I really appreciate it. 

Yes, you're right about using visual block (`<C-v>`), and that's the approach I always use. However I was playing around with mini.align to see what kind of things I can do. Then encountered that situation where I wanted to add spaces between tow columns, wondered if could do it with mini.align, didn't figure it out and ended up here.

And actually your answer was really helpful because I wasn't sure about how to use split or merge or how they work (even though I read the docs), this gives a more concise example about how to use them.

**Reply by echasnovski** [reply]
> And actually your answer was really helpful because I wasn't sure about how to use split or merge or how they work (even though I read the docs), this gives a more concise example about how to use them.

Yeah, 'mini.align' has many possibilities which have to be documented somehow. I tried to make them as accessible as I could.

Have you tried going through [examples](https://github.com/echasnovski/mini.nvim/blob/main/doc/mini-align.txt#L358)?

**Reply by rafamadriz** [reply]
Yes I have gone through examples, it took me going through it a couple of times to understand most of it, but that's fine. I wonder why modifiers adding pre-steps don't have a "hint".

e.g the main modifiers show you this:
  
![Screenshot from 2023-04-30 18-13-40](https://user-images.githubusercontent.com/67771985/235376561-4054bce3-a029-47e1-b0b4-a22a8626db87.png)
	
But nothing for f, i, p, t

**Reply by echasnovski** [reply]
> Yes I have gone through examples, it took me going through it a couple of times to understand most of it, but that's fine. I wonder why modifiers adding pre-steps don't have a "hint".
> ...
> But nothing for f, i, p, t

Mostly to save space. They are shown if active, though. Like if you type `gAip#pt`, it will show both `pair` and `trim` pre-justify steps.
